feat(rentx): adds a library for transaction management using xstate

TheAncientGoat opened the pull request:
Please see the `packages/lib/xstate/README.md` for reasoning behind this PR.

There are a few outstanding items that would round this PR off, that I am not sure is supported by ren-js v2
1. Arbitrary confirmation event listening / restoration for mint transactions on the destination chain (to help check that the transaction is complete)
2. Receipt of the release txHash after burning, and watching that transactions' progress to complete the burn flow
3. Mock implementations of the Lock and Mint chains that be used for testing transaction flows

noiach commented:
Everything looks good - one comment is that I think it's worth simplifying `GatewaySession`. There's a few required fields that could be made optional - e.g. `destAsset`, `expiryTime` and `userAddress` (I think it makes more sense if burns use `destAddress`). Also, RenJS's `lockAndMint.confirmed` now emits a `target` event along side `confirmation`, so it should be possible to remove `destConfsTarget` now as well. Other things might be using existing enums like `RenNetwork.Testnet` instead of `"testnet"` in the examples, which helps users know what values it can accept.

noiach commented:
Also - I think it would be good if the README example was runnable. I think all that is missing is a valid `destAddress` and a `blockchainProviders` (which could use a `HDWalletProvider` for Ethereum to keep the code short).

TheAncientGoat commented:
@noiach - thanks for the feedback, good points, will address them. On point re. `expiryTime` though; won't all gateways expire eventually? I think it might be good to keep it mandatory so that developers are aware of it / set a default precedent so that we don't require breaking flows once expiry kicks in.

TheAncientGoat commented:
@noiach - addressed the feedback; kept `userAddress` seperate from `destinationAddress` because we may want to support minting to an account that won't be signing the transaction and `destAddress` is used for signing when minting, but not when burning.
